client-runtime/src/main/java/com/microsoft/rest/RestProxy.java

```diff
@@ -136,25 +136,38 @@ private HttpRequest createHttpRequest(SwaggerMethodParser methodParser, Object[]
         final Object bodyContentObject = methodParser.body(args);
         if (bodyContentObject != null) {
             String contentType = methodParser.bodyContentType();
+            if (contentType == null || contentType.isEmpty()) {
+                contentType = request.headers().value("Content-Type");
+            }
+            if (contentType == null || contentType.isEmpty()) {
+                if (bodyContentObject instanceof byte[] || bodyContentObject instanceof String) {
+                    contentType = ContentType.APPLICATION_OCTET_STREAM;
+                }
+                else {
+                    contentType = ContentType.APPLICATION_JSON;
+                }
+            }
+
+            request.headers().set("Content-Type", contentType);
+
             if (ContentType.APPLICATION_JSON.equalsIgnoreCase(contentType)) {
```

Comment:
contentType will often be in the form of `application/json; charset=utf-8` so a .contains() test may be preferable